src/app/Library/CrudPanel/Traits/FieldsProtectedMethods.php

```diff
@@ -138,14 +138,15 @@ protected function makeSureFieldHasEntity($field)
 
         //if the name is dot notation we are sure it's a relationship
         if (strpos($field['name'], '.') !== false) {
-            $field['entity'] = $field['name'];
-
-            return $field;
+            $possibleMethodName = Str::endsWith(Str::before($field['name'], '.'), '_id') ? Str::replaceLast('_id', '', Str::before($field['name'], '.')) : Str::before($field['name'], '.');
+            // if it has parameters it's not a relation method.
+            $field['entity'] = $this->modelMethodHasParameters($this->model, $possibleMethodName) ? false : $possibleMethodName.'.'.Str::after($field['name'], '.');
```

Comment:
This line is two long AND does THREE things in one line:
- a declaration (`=`)
- a conditional (`?:`)
- a string manipulation

Definitely too difficult to read & understand without proper attention. 
Brevity is not more important than clarity.

The least we could do is move the string manipulation up one line and give it a self-explanatory name, that way that line is easier to read:

```suggestion
            $giveMeAName = $possibleMethodName.'.'.Str::after($field['name'], '.');
            $field['entity'] = $this->modelMethodHasParameters($this->model, $possibleMethodName) ? false : $giveMeAName;
```